electron/game.ts

```diff
@@ -1,85 +1,89 @@
-import * as path from 'path';
-import { getActiveWindow, Window } from './window';
+import { IpcMain, Rectangle } from 'electron'
+import * as path from 'path'
+import { getActiveWindow, Window } from './window'
 
 const POE_NAMES = [
-    'pathofexile_x64_kg.exe', 'pathofexile_kg.exe',
-    'pathofexile_x64steam.exe', 'pathofexilesteam.exe',
-    'pathofexile_x64.exe', 'pathofexile.exe',
-    'pathofexile_x64_kg', 'pathofexile_kg',
-    'pathofexile_x64steam', 'pathofexilesteam',
-    'pathofexile_x64', 'pathofexile',
-    'wine64-preloader' // linux
-];
+  'pathofexile_x64_kg.exe',
+  'pathofexile_kg.exe',
+  'pathofexile_x64steam.exe',
+  'pathofexilesteam.exe',
+  'pathofexile_x64.exe',
+  'pathofexile.exe',
+  'pathofexile_x64_kg',
+  'pathofexile_kg',
+  'pathofexile_x64steam',
+  'pathofexilesteam',
+  'pathofexile_x64',
+  'pathofexile',
+  'wine64-preloader', // linux
+]
 
-const POE_TITLES = [
-    'Path of Exile'
-];
+const POE_TITLES = ['Path of Exile']
 
-const POE_ALTERNATIVE_TITLES = [
-    'Path of Exile <---> '
-];
+const POE_ALTERNATIVE_TITLES = ['Path of Exile <---> ']
 
 export class Game {
-    private window: Window;
+  private window: Window
 
-    public active?: boolean;
-    public bounds?: Rectangle;
+  public active?: boolean
+  public bounds?: Rectangle
 
-    public update(): boolean {
-        const old = this.toString();
+  public async update(): Promise<boolean> {
+    const old = this.toString()
 
-        const window = getActiveWindow();
-        if (window) {
-            const windowPath = (window.path || '').toLowerCase();
-            const name = path.basename(windowPath);
-            if (POE_NAMES.includes(name)) {
-                const title = window.title();
-                if (POE_TITLES.includes(title) || POE_ALTERNATIVE_TITLES.some(x => title.startsWith(x))) {
-                    this.window = window;
-                    this.active = true;
-                    this.bounds = window.bounds();
-                } else {
-                    this.active = false;
-                }
-            } else {
-                this.active = false;
-            }
+    const window = await getActiveWindow()
+    if (window) {
+      const windowPath = (window.path || '').toLowerCase()
+      const name = path.basename(windowPath)
+      if (POE_NAMES.includes(name)) {
+        const title = window.title()
+        if (POE_TITLES.includes(title) || POE_ALTERNATIVE_TITLES.some((x) => title.startsWith(x))) {
+          this.window = window
+          this.active = true
+          this.bounds = window.bounds()
         } else {
-            this.active = false;
+          this.active = false
         }
-        return old !== this.toString();
+      } else {
+        this.active = false
+      }
+    } else {
+      this.active = false
     }
+    return old !== this.toString()
+  }
 
-    public focus(): void {
-        this.window?.bringToTop();
+  public focus(): void {
+    if (this.window?.bringToTop) {
```

Comment:
Is the `?` here checking for `undefined` values from `getActiveWindow()`? Are undefined windows fatal to the Overlay?

Reply:
Yes exactly, if window would be undefined it would throw an exception on the bringToTop function which is obviously not available on undefined